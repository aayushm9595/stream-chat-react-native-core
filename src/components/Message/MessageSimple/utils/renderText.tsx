import React from 'react';
// @ts-expect-error
import Markdown from '@stream-io/react-native-simple-markdown';
import anchorme from 'anchorme';
import truncate from 'lodash/truncate';

import type { MarkdownStyle } from '../../../../styles/themeConstants';
import type { Message } from '../../../MessageList/utils/insertDates';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../../types/types';

const defaultMarkdownStyles = {
  inlineCode: {
    backgroundColor: '#F3F3F3',
    borderColor: '#dddddd',
    color: 'red',
    fontSize: 13,
    padding: 3,
    paddingHorizontal: 5,
  },
  link: {
    color: 'blue',
    textDecorationLine: 'underline',
  },
  url: {
    color: 'blue',
    textDecorationLine: 'underline',
  },
};

export type RenderTextParams<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  markdownRules: UnknownType;
  markdownStyles: MarkdownStyle;
  message: Message<At, Ch, Co, Ev, Me, Re, Us>;
};

export const renderText = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  params: RenderTextParams<At, Ch, Co, Ev, Me, Re, Us>,
) => {
  const { markdownRules, markdownStyles, message } = params;

  // take the @ mentions and turn them into markdown?
  // translate links
  const { mentioned_users = [], text } = message;

  if (!text) return null;

  let newText = text.trim();
  const urls = anchorme(newText, {
    list: true,
  });

  for (const urlInfo of urls) {
    const displayLink = truncate(urlInfo.encoded.replace(/^(www\.)/, ''), {
      length: 200,
      omission: '...',
    });
    const markdown = `[${displayLink}](${urlInfo.protocol}${urlInfo.encoded})`;
    newText = newText.replace(urlInfo.raw, markdown);
  }

  if (mentioned_users.length) {
    for (let i = 0; i < mentioned_users.length; i++) {
      const username = mentioned_users[i].name || mentioned_users[i].id;
      const markdown = `**@${username}**`;
      const regEx = new RegExp(`@${username}`, 'g');
      newText = newText.replace(regEx, markdown);
    }
  }

  newText = newText.replace(/[<&"'>]/g, '\\$&');
  const styles = {
    ...defaultMarkdownStyles,
    ...markdownStyles,
  };

  return (
    <Markdown rules={markdownRules} styles={styles}>
      {newText}
    </Markdown>
  );
};
