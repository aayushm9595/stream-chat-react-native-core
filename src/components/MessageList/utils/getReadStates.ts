import {
  InsertDatesResponse,
  isDateSeparator,
  Message,
} from '../utils/insertDates';

import type { ImmutableDate } from 'seamless-immutable';
import type { UserResponse } from 'stream-chat';

import type { ChannelContextValue } from '../../../contexts/channelContext/ChannelContext';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';

export const getReadStates = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  messages: InsertDatesResponse<At, Ch, Co, Ev, Me, Re, Us>,
  read?: ChannelContextValue<At, Ch, Co, Ev, Me, Re, Us>['read'],
) => {
  const readData = messages.reduce((acc, cur) => {
    if (!isDateSeparator(cur) && cur.id) {
      acc[cur.id] = [];
    }
    return acc;
  }, {} as { [key: string]: UserResponse<Us>[] });

  const filteredMessagesReversed = messages
    .filter((msg) => !isDateSeparator(msg) && msg.updated_at)
    .reverse() as Array<
    Message<At, Ch, Co, Ev, Me, Re, Us> & {
      updated_at: string | ImmutableDate;
    }
  >;

  if (read) {
    /**
     * Channel read state is stored by user
     */
    for (const readState of Object.values(read)) {
      /**
       * If no last read continue
       */
      if (!readState.last_read) {
        continue;
      }

      /**
       * Array is in reverse order so newest message is at 0,
       * we find the index of the first message that is older
       * than the last read and then set last read to that, or
       * if there are no newer messages, the first message is
       * last read message.
       */
      const userLastReadMsgId = filteredMessagesReversed.find(
        (msg) => msg.updated_at < readState.last_read,
      )?.id;

      /**
       * If there there is a last read message add the user
       * to the array of last reads for that message
       */
      if (userLastReadMsgId) {
        readData[userLastReadMsgId] = [
          ...readData[userLastReadMsgId],
          readState.user as UserResponse<Us>,
        ];
      }
    }
  }

  return readData;
};
