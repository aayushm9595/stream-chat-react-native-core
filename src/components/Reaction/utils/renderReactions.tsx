import React from 'react';
import { Text } from 'react-native';

import type { ReactionResponse } from 'stream-chat';

import type { LatestReactions, Reaction } from '../ReactionList';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';

export const renderReactions = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  reactions: LatestReactions<At, Ch, Co, Me, Re, Us>,
  supportedReactions: Reaction[],
) => {
  const reactionsByType: {
    [key: string]: ReactionResponse<Re, Us>[];
  } = {};
  reactions?.map((reaction) => {
    if (!reactionsByType[reaction.type]) {
      return (reactionsByType[reaction.type] = [reaction]);
    }
    return (reactionsByType[reaction.type] = [
      ...reactionsByType[reaction.type],
      reaction,
    ]);
  });

  const emojiDataByType: { [key: string]: Reaction } = {};
  const reactionTypes = supportedReactions.map((supportedReaction) => {
    emojiDataByType[supportedReaction.id] = supportedReaction;
    return supportedReaction.id;
  });

  return Object.keys(reactionsByType).map((type) =>
    reactionTypes.indexOf(type) > -1 ? (
      <Text key={type} testID={type}>
        {emojiDataByType[type].icon}
      </Text>
    ) : null,
  );
};
