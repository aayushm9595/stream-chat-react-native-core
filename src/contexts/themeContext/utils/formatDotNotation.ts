import lodashGet from 'lodash/get';
import isPlainObject from 'lodash/isPlainObject';
import merge from 'lodash/merge';
import lodashSet from 'lodash/set';

import type { ThemeType } from './replaceCssShorthand';

import type { Theme } from '../../../styles/themeConstants';

export type FormatDotNotationParams = {
  formattedStyle: string | number | boolean | ThemeType;
  modifiedTheme: Theme;
};

export const formatDotNotation = ({
  formattedStyle,
  modifiedTheme,
}: FormatDotNotationParams) => {
  if (isPlainObject(formattedStyle)) {
    const themeDiff = {};
    for (const k in formattedStyle as ThemeType) {
      if (lodashGet(modifiedTheme, k)) {
        merge(themeDiff, lodashSet({}, k, (formattedStyle as ThemeType)[k]));
      } else {
        throw Error(`Unknown theme key ${k}`);
      }
    }
    return themeDiff;
  }
  return modifiedTheme;
};
