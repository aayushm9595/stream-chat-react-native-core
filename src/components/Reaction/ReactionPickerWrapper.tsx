import React, { PropsWithChildren, useEffect, useRef, useState } from 'react';
import { Dimensions, TouchableOpacity } from 'react-native';

import { ReactionPicker as ReactionPickerDefault } from './ReactionPicker';

import { emojiData as emojiDataDefault } from '../../utils/utils';

import type { LatestReactions, Reaction } from './ReactionList';
import type { ReactionPickerProps } from './ReactionPicker';

import type {
  Alignment,
  MessageWithDates,
} from '../../contexts/messagesContext/MessagesContext';
import type { Message } from '../../components/MessageList/utils/insertDates';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../types/types';

export type ReactionPickerWrapperProps<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  alignment: Alignment;
  customMessageContent: boolean;
  dismissReactionPicker: () => void;
  handleReaction: (id: Reaction['id']) => void;
  hideReactionCount: boolean;
  hideReactionOwners: boolean;
  message: Message<At, Ch, Co, Ev, Me, Re, Us>;
  offset: { left: number; right: number; top: number };
  openReactionPicker: () => void;
  reactionPickerVisible: boolean;
  /**
   * e.g.,
   * [
   *  {
   *    id: 'like',
   *    icon: 'üëç',
   *  },
   *  {
   *    id: 'love',
   *    icon: '‚ù§Ô∏èÔ∏è',
   *  },
   *  {
   *    id: 'haha',
   *    icon: 'üòÇ',
   *  },
   *  {
   *    id: 'wow',
   *    icon: 'üòÆ',
   *  },
   * ]
   */
  supportedReactions: Reaction[];
  /**
   * @deprecated
   * emojiData is deprecated. But going to keep it for now
   * to have backward compatibility. Please use supportedReactions instead.
   * TODO: Remove following prop in 1.x.x
   */
  emojiData?: Reaction[];
  ReactionPicker?: React.ComponentType<
    ReactionPickerProps<At, Ch, Co, Me, Re, Us>
  >;
};

export const ReactionPickerWrapper = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  props: PropsWithChildren<
    ReactionPickerWrapperProps<At, Ch, Co, Ev, Me, Re, Us>
  >,
) => {
  const {
    alignment,
    children,
    customMessageContent,
    dismissReactionPicker,
    emojiData = emojiDataDefault,
    handleReaction,
    hideReactionCount = false,
    hideReactionOwners = false,
    message,
    offset = { left: 30, right: 10, top: 40 },
    openReactionPicker,
    ReactionPicker = ReactionPickerDefault,
    reactionPickerVisible,
    supportedReactions = emojiDataDefault,
  } = props;

  const messageContainer = useRef<TouchableOpacity>(null);
  const [rpLeft, setRPLeft] = useState<number>();
  const [rpRight, setRPRight] = useState<number>();
  const [rpTop, setRPTop] = useState<number>();

  useEffect(() => {
    if (reactionPickerVisible) {
      setReactionPickerPosition();
    }
  }, [reactionPickerVisible]);

  const setReactionPickerPosition = () => {
    setTimeout(
      () => {
        if (messageContainer.current) {
          messageContainer.current.measureInWindow((x, y, width) => {
            setRPLeft(alignment === 'left' ? x - 10 + offset.left : undefined);
            setRPRight(
              alignment === 'right'
                ? Math.round(Dimensions.get('window').width) -
                    (x + width + offset.right)
                : undefined,
            );
            setRPTop(y - 60 + offset.top);
          });
        }
      },
      customMessageContent ? 10 : 0,
    );
  };

  return (
    <TouchableOpacity
      onPress={openReactionPicker}
      ref={messageContainer}
      testID='reaction-picker-wrapper'
    >
      {children}
      <ReactionPicker<At, Ch, Co, Me, Re, Us>
        {...props}
        handleDismiss={dismissReactionPicker}
        handleReaction={handleReaction}
        hideReactionCount={hideReactionCount}
        hideReactionOwners={hideReactionOwners}
        latestReactions={
          message.latest_reactions as LatestReactions<At, Ch, Co, Me, Re, Us>
        }
        reactionCounts={
          message.reaction_counts as MessageWithDates<
            At,
            Ch,
            Co,
            Me,
            Re,
            Us
          >['reaction_counts']
        }
        reactionPickerVisible={reactionPickerVisible}
        rpLeft={rpLeft}
        rpRight={rpRight}
        rpTop={rpTop}
        supportedReactions={supportedReactions || emojiData}
      />
    </TouchableOpacity>
  );
};
