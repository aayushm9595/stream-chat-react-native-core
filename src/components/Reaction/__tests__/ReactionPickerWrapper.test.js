import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';

import { generateUser } from 'mock-builders/generator/user';
import { getTestClientWithUser } from 'mock-builders/mock';

import { ReactionPickerWrapper } from '../ReactionPickerWrapper';

import { Chat } from '../../Chat/Chat';

import { emojiData } from '../../../utils/utils';

const defaultProps = {
  alignment: 'left',
  message: {
    latestReactions: [
      {
        created_at: '2019-11-28T11:14:20.155817Z',
        message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
        score: 1,
        type: 'sad',
        updated_at: '2019-11-28T11:14:20.155817Z',
        user: {
          banned: false,
          created_at: '2019-08-26T10:49:19.667963Z',
          id: 'bender',
          image: 'https://bit.ly/321RmWb',
          last_active: '2020-01-06T08:42:44.097139Z',
          name: 'Bender',
          online: false,
          role: 'user',
          updated_at: '2020-01-06T08:42:44.106042Z',
        },
        user_id: 'bender',
      },
      {
        created_at: '2019-11-28T11:13:54.845187Z',
        message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
        score: 1,
        type: 'haha',
        updated_at: '2019-11-28T11:13:54.845187Z',
        user: {
          banned: false,
          created_at: '2019-08-26T10:49:19.667963Z',
          id: 'bender',
          image: 'https://bit.ly/321RmWb',
          last_active: '2020-01-06T08:42:44.097139Z',
          name: 'Bender',
          online: false,
          role: 'user',
          updated_at: '2020-01-06T08:42:44.106042Z',
        },
        user_id: 'bender',
      },
      {
        created_at: '2019-11-28T10:32:58.981307Z',
        message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
        score: 1,
        type: 'like',
        updated_at: '2019-11-28T10:32:58.981307Z',
        user: {
          banned: false,
          created_at: '2019-08-26T10:49:19.667963Z',
          id: 'bender',
          image: 'https://bit.ly/321RmWb',
          last_active: '2020-01-06T08:42:44.097139Z',
          name: 'Bender',
          online: false,
          role: 'user',
          updated_at: '2020-01-06T08:42:44.106042Z',
        },
        user_id: 'bender',
      },
      {
        created_at: '2019-11-26T22:32:21.975407Z',
        message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
        score: 1,
        type: 'sad',
        updated_at: '2019-11-26T22:32:21.975407Z',
        user: {
          banned: false,
          created_at: '2019-04-02T13:26:55.274436Z',
          id: 'billowing-firefly-8',
          image:
            'https://stepupandlive.files.wordpress.com/2014/09/3d-animated-frog-image.jpg',
          last_active: '2020-01-06T16:47:25.692477Z',
          name: 'Billowing firefly',
          online: true,
          role: 'user',
          updated_at: '2020-01-06T16:47:25.696327Z',
        },
        user_id: 'billowing-firefly-8',
      },
      {
        created_at: '2019-11-26T22:31:51.31847Z',
        message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
        score: 1,
        type: 'like',
        updated_at: '2019-11-26T22:31:51.31847Z',
        user: {
          banned: false,
          created_at: '2019-04-02T13:26:55.274436Z',
          id: 'billowing-firefly-8',
          image:
            'https://stepupandlive.files.wordpress.com/2014/09/3d-animated-frog-image.jpg',
          last_active: '2020-01-06T16:47:25.692477Z',
          name: 'Billowing firefly',
          online: true,
          role: 'user',
          updated_at: '2020-01-06T16:47:25.696327Z',
        },
        user_id: 'billowing-firefly-8',
      },
    ],
    reaction_counts: {
      haha: 1,
      like: 2,
      sad: 2,
    },
  },
  reactionPickerVisible: true,
};

describe('ReactionPickerWrapper', () => {
  const clientUser = generateUser();
  let chatClient;

  const getComponent = (props = {}) => (
    <Chat client={chatClient}>
      <ReactionPickerWrapper {...props} />
    </Chat>
  );

  beforeEach(async () => {
    chatClient = await getTestClientWithUser(clientUser);
  });

  it('should render left aligned ReactionPickerWrapper', async () => {
    const props = { ...defaultProps, handleReaction: jest.fn() };
    const { getByTestId, queryByTestId, toJSON } = render(getComponent(props));

    await waitFor(() => {
      expect(queryByTestId('reaction-picker-wrapper')).toBeTruthy();
      expect(queryByTestId('reaction-picker')).toBeTruthy();
      emojiData.forEach((emoji) => {
        expect(queryByTestId(`${emoji.id}-reaction`)).toBeTruthy();
        expect(
          queryByTestId(
            `${emoji.id}-${
              defaultProps.message.reaction_counts[emoji.id] || 'count'
            }`,
          ),
        ).toBeTruthy();
        expect(
          getByTestId(
            `${emoji.id}-${
              defaultProps.message.reaction_counts[emoji.id] || 'count'
            }`,
          ),
        ).toHaveTextContent(
          defaultProps.message.reaction_counts[emoji.id] || '',
        );
      });
      expect(props.handleReaction).toHaveBeenCalledTimes(0);
    });

    emojiData.forEach((emoji) => {
      fireEvent.press(getByTestId(`${emoji.id}-reaction`));
    });

    await waitFor(() => {
      expect(props.handleReaction).toHaveBeenCalledTimes(emojiData.length);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render right aligned ReactionPickerWrapper', async () => {
    const props = {
      ...defaultProps,
      alignment: 'right',
      handleReaction: jest.fn(),
    };
    const { getByTestId, queryByTestId, toJSON } = render(getComponent(props));

    await waitFor(() => {
      expect(queryByTestId('reaction-picker-wrapper')).toBeTruthy();
      expect(queryByTestId('reaction-picker')).toBeTruthy();
      emojiData.forEach((emoji) => {
        expect(queryByTestId(`${emoji.id}-reaction`)).toBeTruthy();
        expect(
          queryByTestId(
            `${emoji.id}-${
              defaultProps.message.reaction_counts[emoji.id] || 'count'
            }`,
          ),
        ).toBeTruthy();
        expect(
          getByTestId(
            `${emoji.id}-${
              defaultProps.message.reaction_counts[emoji.id] || 'count'
            }`,
          ),
        ).toHaveTextContent(
          defaultProps.message.reaction_counts[emoji.id] || '',
        );
      });
      expect(props.handleReaction).toHaveBeenCalledTimes(0);
    });

    emojiData.forEach((emoji) => {
      fireEvent.press(getByTestId(`${emoji.id}-reaction`));
    });

    await waitFor(() => {
      expect(props.handleReaction).toHaveBeenCalledTimes(emojiData.length);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render not visible ReactionPickerWrapper', async () => {
    const props = {
      ...defaultProps,
      openReactionPicker: jest.fn(),
      reactionPickerVisible: false,
    };
    const { getByTestId, queryByTestId, toJSON } = render(getComponent(props));

    await waitFor(() => {
      expect(queryByTestId('reaction-picker-wrapper')).toBeTruthy();
      expect(queryByTestId('reaction-picker')).toBeFalsy();
      emojiData.forEach((emoji) => {
        expect(queryByTestId(`${emoji.id}-reaction`)).toBeFalsy();
        expect(
          queryByTestId(
            `${emoji.id}-${
              defaultProps.message.reaction_counts[emoji.id] || 'count'
            }`,
          ),
        ).toBeFalsy();
        expect(props.openReactionPicker).toHaveBeenCalledTimes(0);
      });
    });

    fireEvent.press(getByTestId('reaction-picker-wrapper'));

    await waitFor(() => {
      expect(props.openReactionPicker).toHaveBeenCalledTimes(1);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
