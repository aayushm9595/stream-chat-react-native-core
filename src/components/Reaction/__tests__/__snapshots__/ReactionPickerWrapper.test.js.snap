// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionPickerWrapper should render left aligned ReactionPickerWrapper 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="reaction-picker-wrapper"
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="reaction-picker"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 60,
              "paddingHorizontal": 20,
            },
            Object {
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="like"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="like-reaction"
          >
            👍
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="like-2"
          >
            2
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="love"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="love-reaction"
          >
            ❤️️
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="love-count"
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="haha"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="haha-reaction"
          >
            😂
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="haha-1"
          >
            1
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="wow"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="wow-reaction"
          >
            😮
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="wow-count"
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="sad"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="sad-reaction"
          >
            😔
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="sad-2"
          >
            2
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="angry"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="angry-reaction"
          >
            😠
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="angry-count"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`ReactionPickerWrapper should render not visible ReactionPickerWrapper 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="reaction-picker-wrapper"
/>
`;

exports[`ReactionPickerWrapper should render right aligned ReactionPickerWrapper 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="reaction-picker-wrapper"
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    testID="reaction-picker"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 60,
              "paddingHorizontal": 20,
            },
            Object {
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="like"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="like-reaction"
          >
            👍
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="like-2"
          >
            2
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="love"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="love-reaction"
          >
            ❤️️
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="love-count"
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="haha"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="haha-reaction"
          >
            😂
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="haha-1"
          >
            1
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="wow"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="wow-reaction"
          >
            😮
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="wow-count"
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="sad"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="sad-reaction"
          >
            😔
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="sad-2"
          >
            2
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "marginTop": -5,
              },
            ]
          }
          testID="angry"
        >
          <View
            style={
              Object {
                "height": 18,
                "width": 18,
              }
            }
          />
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "marginVertical": 5,
                },
              ]
            }
            testID="angry-reaction"
          >
            😠
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 10,
                  "fontWeight": "bold",
                },
              ]
            }
            testID="angry-count"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
